overlord/ifacestate/ifacestate.go

```diff
@@ -107,14 +107,19 @@ func AutoConnect(st *state.State, change *state.Change, autoConnectTask *state.T
 	return connect(st, change, plugSnap, plugName, slotSnap, slotName, autoConnectTask)
 }
 
+// Reconnect returns a set of tasks for reconnecting an interface as part of snap installation.
+func Reconnect(st *state.State, change *state.Change, reconnectTask *state.Task, plugSnap, plugName, slotSnap, slotName string) (*state.TaskSet, error) {
+	return connect(st, change, plugSnap, plugName, slotSnap, slotName, reconnectTask)
```

Comment:
Is there a benefit in having the exact same method with two different names and two different argument orders?

Reply:
Fair point. This is just to make the code more readable on the caller side and to emphasize that these are slightly different operations, although the difference is realized internally in `connect` based on properties of the autoconnectTask/reconnectTask.

I've adjusted the argument order of `connect` though so the order is same now.

Reply:
They can't possibly be slightly different operations if they take exactly the same arguments and one of them just calls the other. There's no room for being different.

Reply:
I can reduce them to a single method if we agree on a name.. I couldn't find a good name and having `Reconnect` and `Autoconnect` helped avoid the naming problem :blush: